<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlaybackWidget</class>
 <widget class="QWidget" name="PlaybackWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Playback</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="fileGroupBox">
     <property name="title">
      <string>Recording File</string>
     </property>
     <layout class="QHBoxLayout" name="fileLayout">
      <item>
       <widget class="QLineEdit" name="filePathLineEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>No file selected</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="browseFileButton">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="reloadFileButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Reload</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="controlsGroupBox">
     <property name="title">
      <string>Playback Controls</string>
     </property>
     <layout class="QVBoxLayout" name="controlsMainLayout">
      <item>
       <layout class="QHBoxLayout" name="buttonsLayout">
        <item>
         <widget class="QPushButton" name="playButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Play</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/icons/play.png</normaloff>:/icons/play.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pauseButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Pause</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/icons/pause.png</normaloff>:/icons/pause.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="stopButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/icons/stop.png</normaloff>:/icons/stop.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="controlsSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="loopCheckBox">
          <property name="text">
           <string>Loop Playback</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="progressLayout">
        <item>
         <widget class="QLabel" name="currentTimeLabel">
          <property name="text">
           <string>00:00</string>
          </property>
          <property name="styleSheet">
           <string>font-family: monospace;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="progressSlider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="totalTimeLabel">
          <property name="text">
           <string>00:00</string>
          </property>
          <property name="styleSheet">
           <string>font-family: monospace;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="speedGroupBox">
     <property name="title">
      <string>Playback Speed</string>
     </property>
     <layout class="QHBoxLayout" name="speedLayout">
      <item>
       <widget class="QLabel" name="speedLabel">
        <property name="text">
         <string>Speed:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="speedSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="speedValueLabel">
        <property name="text">
         <string>1.0x</string>
        </property>
        <property name="minimumSize">
         <size>
          <width>40</width>
          <height>0</height>
         </size>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="speedSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="resetSpeedButton">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="informationGroupBox">
     <property name="title">
      <string>Recording Information</string>
     </property>
     <layout class="QGridLayout" name="informationLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="fileFormatLabel">
        <property name="text">
         <string>File Format:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="fileFormatValue">
        <property name="text">
         <string>-</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="totalEventsLabel">
        <property name="text">
         <string>Total Events:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="totalEventsValue">
        <property name="text">
         <string>-</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="durationLabel">
        <property name="text">
         <string>Duration:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="durationValue">
        <property name="text">
         <string>-</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="createdLabel">
        <property name="text">
         <string>Created:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="createdValue">
        <property name="text">
         <string>-</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="eventsPreviewGroupBox">
     <property name="title">
      <string>Events Preview</string>
     </property>
     <layout class="QVBoxLayout" name="eventsPreviewLayout">
      <item>
       <widget class="QTableWidget" name="eventsPreviewTableWidget">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <column>
         <property name="text">
          <string>Index</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Time</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Type</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Details</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
